src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -141,6 +142,34 @@ def _create_dev_app(self, diff: DiffResult) -> None:
             except ProgrammingError as err:
                 generic_sql_error_handler(err)
 
+            self._execute_post_deploy_hooks()
+
+    def _execute_sql_script(self, sql_script_path):
+        with open(sql_script_path) as f:
+            sql_script = f.read()
+            try:
+                if self.application_warehouse:
+                    self._execute_query(f"use warehouse {self.application_warehouse}")
+                if self._conn.database:
+                    self._execute_query(f"use database {self._conn.database}")
+                self._execute_queries(sql_script)
+            except ProgrammingError as err:
+                generic_sql_error_handler(err)
+
+    def _execute_post_deploy_hooks(self):
+        post_deploy_script_hooks = self.app_post_deploy_hooks
+        if post_deploy_script_hooks:
+            for hook in post_deploy_script_hooks:
+                if hook.sql_script:
+                    cc.step(
```

Comment:
consistency: consider using a phase here with one step per script. That's what we do for codegen.